lldb/source/Plugins/Language/Swift/SwiftUnsafeTypes.cpp

```diff
@@ -423,9 +423,80 @@ bool lldb_private::formatters::swift::UnsafeTypeSummaryProvider(
   return true;
 }
 
+static std::vector<ValueObjectSP>
+ExtractChildrenFromSwiftPointerValueObject(ValueObjectSP valobj_sp,
+                                           SwiftUnsafeType &unsafe_ptr) {
+  if (!valobj_sp)
+    return {};
+
+  lldb::ProcessSP process_sp(valobj_sp->GetProcessSP());
+
+  if (!process_sp)
+    return {};
+
+  const addr_t start_addr = unsafe_ptr.GetStartAddress();
+  const size_t num_children = unsafe_ptr.GetCount();
+  const CompilerType element_type = unsafe_ptr.GetElementType();
+
+  auto stride = element_type.GetByteStride(process_sp.get());
+  if (!stride)
+    return {};
+
+  auto element_stride = *stride;
+  size_t buffer_size = num_children * element_stride;
+  if (buffer_size > 512 * 1024 * 1024) {
+    LLDB_LOG(GetLog(LLDBLog::DataFormatters),
+             "Suspiciously large object: num_children={0}, stride={1}",
+             num_children, element_stride);
+    return {};
+  }
+
+  std::unique_ptr<lldb_private::WritableDataBuffer> buffer_up(
+      new DataBufferHeap(buffer_size, 0));
+  Status error;
+  size_t read_bytes = process_sp->ReadMemory(start_addr, buffer_up->GetBytes(),
+                                             buffer_size, error);
+
+  if (!read_bytes || error.Fail())
+    return {};
+
+  auto ptr_size = process_sp->GetAddressByteSize();
+  auto order = process_sp->GetByteOrder();
+  DataExtractor buffer_data(buffer_up->GetBytes(), buffer_up->GetByteSize(),
+                            order, ptr_size);
+
+  std::vector<ValueObjectSP> children;
+  auto exe_ctx_ref = valobj_sp->GetExecutionContextRef();
+  // UnsafePointer/UnsafeMutablePointer have a `pointee` property.
+  if (unsafe_ptr.HasPointee()) {
+    DataExtractor data(buffer_data, 0, element_stride);
+    children.push_back(ValueObject::CreateValueObjectFromData(
+        "pointee", data, exe_ctx_ref, element_type));
+    return children;
+  }
+
+  for (size_t i = 0; i < num_children; i++) {
+    StreamString idx_name;
+    idx_name.Printf("[%zu]", i);
```

Comment:
Using one of the string stream objects we have in lldb and llvm might be less error prone than printf

Reply:
what I mean is using an `operator<<`

Reply:
`operator<< ` is not implemented for `lldb_private::StreamString` and `size_t`, maybe that's why it was a printf in the first place